modules/core/include/opencv2/core/hal/hal.hpp

```diff
@@ -95,11 +95,15 @@ CV_EXPORTS void fastAtan32f(const float* y, const float* x, float* dst, int n, b
 CV_EXPORTS void fastAtan64f(const double* y, const double* x, double* dst, int n, bool angleInDegrees);
 CV_EXPORTS void magnitude32f(const float* x, const float* y, float* dst, int n);
 CV_EXPORTS void magnitude64f(const double* x, const double* y, double* dst, int n);
+CV_EXPORTS void magnitudeSqr32f(const float* x, const float* y, float* dst, int n);
+CV_EXPORTS void magnitudeSqr64f(const double* x, const double* y, double* dst, int n);
 CV_EXPORTS void sqrt32f(const float* src, float* dst, int len);
 CV_EXPORTS void sqrt64f(const double* src, double* dst, int len);
 CV_EXPORTS void invSqrt32f(const float* src, float* dst, int len);
 CV_EXPORTS void invSqrt64f(const double* src, double* dst, int len);
 
+CV_EXPORTS void sqr64f(const double* src, double* dst, int len);
```

Comment:
This function is not used because ipp branch is commented, maybe we can remove it? Or add `sqr32f` and  `cv::sqr()` for completeness?